### DISCUSSION TITLE
target attribute doesn't seem to be working as it should

### DISCUSSION THREAD

**Discussion by ItWorksOnMyMachine** [webpack/webpack#13803] [General]
# Bug report

**What is the current behavior?**
I got a BREAKING CHANGE notice after upgrading from v4. It was related to polyfills.  I resolved it by adding the target: "web" attribute to my config. Everything built and ran fine.  Then, I got to the part of my upgrade where I have to support IE11 and realized there was a note about using ["web", "es5"] or setting target to browserslist.  I found that doing either of those started throwing the polyfills BREAKING CHANGE error again.  I then returned to just "web", however, I set it as an array instead of string, so target: ["web"].  This still caused the polyfills breaking change notice.  Once I returned to target: "web" everything worked again.

**If the current behavior is a bug, please provide the steps to reproduce.**
See above

**What is the expected behavior?**
I would expect that target: "web" and target: ["web"] would work the exact same.  I would also expect target: ["web", "es5"] or target: "browserslist" to compile without complaining about polyfills that nodejs needs.  

**Other relevant information:**
webpack version:  5.43.0
Node.js version: 16.4.2
Operating System: Windows 10
Additional tools: Aurelia-based application.

**Comment by alexander-akait** [comment]
`target: 'web'` and `target: ['web']` is the same https://github.com/webpack/webpack/blob/main/lib/config/target.js#L100

**Reply by ItWorksOnMyMachine** [reply]
Sorry @alexander-akait, I'm not following.  Is there another issue open on this?  It doesn't behave the same.  If I use "web" everything builds fine.  If I use ["web"] I get these errors:

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "punycode": require.resolve("punycode/") }'
        - install 'punycode'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "punycode": false }

**Reply by alexander-akait** [reply]
You will have the same error, not matter is target array or not, please provide example how we can reproduce

**Reply by ItWorksOnMyMachine** [reply]
I'm working on a project to reproduce it for you.  However, adding target: "web" is how I got those errors to go away.  Not sure why it works that way, I don't know the code for webpack.  I made that change because I found several other threads in projects that use webpack that also said that's how they eliminated the errors.  Unfortunately, I can't share the actual project as it's a very large and proprietary project.

**Reply by ItWorksOnMyMachine** [reply]
I think this may actually be an issue with the aurelia-webpack plugin.

**Reply by alexander-akait** [reply]
> I think this may actually be an issue with the aurelia-webpack plugin.

I think the same, if you provide example I will look

**Reply by ItWorksOnMyMachine** [reply]
I opened an issue with the aurelia-webpack plugin here: https://github.com/aurelia/webpack-plugin/issues/187
There's a repro repo here: https://github.com/ItWorksOnMyMachine/au-test-20210712

It looks to me like aurelia-bootstrapper is misunderstanding which PAL it should load and is loading the nodejs PAL in order to do server-side rendering.  I'm not sure why this is happening however, as it doesn't look at the webpack "target" config attribute.

**Reply by ItWorksOnMyMachine** [reply]
BTW, thanks for you help with this and with the css-loader issue.
